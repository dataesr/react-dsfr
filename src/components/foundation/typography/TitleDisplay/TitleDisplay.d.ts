import * as React from 'react';

export type TitleDisplayChildren = string | React.ReactNode;

export type TitleDisplayClassName = string | Object | any[];

export type TitleDisplayAs = "h1" | "h2" | "h3" | "h4";

export type TitleDisplaySize = "xl" | "lg" | "md" | "sm" | "xs";

export interface TitleDisplayProps {
    children: TitleDisplayChildren;
    className?: TitleDisplayClassName;
    /**
     * html tag to render
     */
    as?: TitleDisplayAs;
    size: TitleDisplaySize;
}

declare const TitleDisplay: React.FC<TitleDisplayProps>;

export default TitleDisplay;

