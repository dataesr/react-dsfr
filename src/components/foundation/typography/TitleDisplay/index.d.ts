import type TitleDisplay from "./TitleDisplay"

export type { TitleDisplay }
