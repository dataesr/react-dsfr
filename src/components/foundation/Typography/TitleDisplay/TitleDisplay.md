### Alternative titles

```jsx
<TitleDisplay as="h1" size="xl">Title xl</TitleDisplay>
```

```jsx
<TitleDisplay as="h1" size="lg">Title lg</TitleDisplay>
```

```jsx
<TitleDisplay as="h1" size="md">Title md</TitleDisplay>
```

```jsx
<TitleDisplay as="h1" size="sm">Title sm</TitleDisplay>
```

```jsx
<TitleDisplay as="h1" size="xs">Title xs</TitleDisplay>
```
