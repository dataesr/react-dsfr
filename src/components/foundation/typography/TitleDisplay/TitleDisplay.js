import PropTypes from 'prop-types';
import classnames from 'classnames';
import { CLASS_NAME_TYPE } from '../../../../utils/variables';

/**
 *
 * @visibleName TitleDisplay
 */
const TitleDisplay = ({
  as, size, children, className,
}) => {
  const HtmlTag = `${as}`;
  return (
    <HtmlTag className={classnames({
      className,
      [`rf-display-${size}`]: size,
    })}
    >
      {children}
    </HtmlTag>
  );
};

TitleDisplay.propTypes = {
  children: PropTypes.string.isRequired,
  /**
   * One of: string, object
   */
  className: CLASS_NAME_TYPE,
  /**
  * html tag to render
  */
  as: PropTypes.oneOf(['h1', 'h2', 'h3', 'h4']),
  size: PropTypes.oneOf(['xl', 'lg', 'md', 'sm', 'xs']).isRequired,
};

TitleDisplay.defaultProps = {
  as: 'h1',
  className: '',
};

export default TitleDisplay;
