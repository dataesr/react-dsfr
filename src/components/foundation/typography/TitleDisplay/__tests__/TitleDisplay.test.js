import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import TitleDisplay from '../TitleDisplay';

const initialProps = {
  as: 'h2',
  size: 'xl',
  children: 'Title Alt xl',
};

describe('<TitleDisplay />', () => {
  it('renders correctly', () => {
    const component = renderer
      .create(<TitleDisplay {...initialProps}>{initialProps.children}</TitleDisplay>)
      .toJSON();
    expect(component).toMatchSnapshot();
  });
});
