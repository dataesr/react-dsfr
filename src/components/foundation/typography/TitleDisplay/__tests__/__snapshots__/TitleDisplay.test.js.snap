// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TitleDisplay /> renders correctly 1`] = `
<h2
  className="rf-display-xl"
>
  Title Alt xl
</h2>
`;
